build/gulpfile.extensions.js

```diff
@@ -40,23 +40,23 @@ const tasks = compilations.map(function (tsconfigFile) {
 	const tsOptions = _.assign({}, tsconfig.extends ? require(path.join(extensionsPath, relativeDirname, tsconfig.extends)).compilerOptions : {}, tsconfig.compilerOptions);
 	tsOptions.verbose = false;
 	tsOptions.sourceMap = true;
+	const outDir = tsOptions.outDir || 'out';
 
 	const name = relativeDirname.replace(/\//g, '-');
 
 	const root = path.join('extensions', relativeDirname);
 	const srcBase = path.join(root, 'src');
 	const src = path.join(srcBase, '**');
-	const out = path.join(root, 'out');
+	const out = path.join(root, outDir);
 	const baseUrl = getBaseUrl(out);
-
-	let headerId, headerOut;
+		let headerId, headerOut;
 	let index = relativeDirname.indexOf('/');
 	if (index < 0) {
 		headerId = 'vscode.' + relativeDirname;
-		headerOut = 'out';
+		headerOut = outDir;
```

Comment:
Does anyone know of a projects which would generate files to headerOut directory?